doc/release/1.16.0-notes.rst

```diff
@@ -113,6 +117,23 @@ Previously, only the ``dims`` keyword argument was accepted
 for specification of the shape of the array to be used
 for unraveling. ``dims`` remains supported, but is now deprecated.
 
+multi-field views return a view instead of a copy
+-------------------------------------------------
+Indexing a structured array with multiple fields, e.g.,
+``arr[['f1', 'f3']]``, returns a view into the original array instead of a
+copy. The returned view will often have extra padding bytes corresponding to
+intervening fields in the original array, unlike before, which will
+affect code such as ``arr[['f1', 'f3']].view('float64')``. This change has
+been planned since numpy 1.7 and such operations have emitted
+``FutureWarnings`` since then and more since 1.12.
+
+To help users update their code to account for these changes, a number of
+functions have been added to the ``numpy.lib.recfunctions`` module which
+safely allow such operations. For instance, the above view can be achieved
+with ``structured_to_unstructured(arr[['f1', 'f3']], dtype='float64')``.
```

Comment:
It would be neat if it did produce a view, but right now this produces a copy not a view, right?

Reply:
This particular example produces a copy, because even before this PR a copy was unavoidable.

However, it was actually your suggestion to rework the code so `structured_to_unstructured` returns a view in many cases, for instance `structured_to_unstructured(arr[['f1', 'f2', 'f3']], dtype='float64')` actually makes no copies.

Reply:
This sentence should be reworked to avoid the word view though...

Reply:
> for instance structured_to_unstructured(arr[['f1', 'f2', 'f3']], dtype='float64') actually makes no copies.

Is that a statement about what actually happens in master, or just what I was suggesting?

Reply:
For clarity, let's say we're dealing with the array `arr = np.ones(3, dtype='f8,f8,f8')`. Then `arr[['f1', 'f3']]` returns a 24-byte structured array where each element is organized as `FxF` where `F` is 8 bytes of float memory, and x is 8 bytes of padding. The 3-element array as a whole is `FxFFxFFxF` (72 bytes). 

Is it possible to view this as an unstructured float64 array with the right stride? I suppose this particular 3-field array might be viewed with shape (3, 2) and stride (24, 16). `np.ndarray((3,2), strides=(24, 16), dtype='f8', buffer=arr)` works.

But it seems it would be quite an involved computation to determine more generally whether appropriate strides exist, given arbitrary field offsets. Fo three fields it is always possible, but with 4 fields it is not, eg `np.ones(3, 'f8,f8,f8,f8')[['f0', 'f1, 'f3']]` cannot be viewed as an unstructured array.

My feeling is it's too difficult to do the calculation generally, it needs some complex gcd computation.

Reply:
> But it seems it would be quite an involved computation to determine more generally whether appropriate strides exist, given arbitrary field offsets

The computation is straightforward:
```python
if len(offsets) == 0:
    stride = 0
    offset = 0
elif len(offsets) == 1:
    stride = 0
    offset = offsets[0]
else:
    offset = offsets[0]
    stride = offsets[1] - offsets[0]
    for i, o in enumerate(offsets[2:],2):
        if o != offset + stride * i:
            raise NotViewable
return offset, stride
```

Which should give enough information to be passed to `as_strided` somehow

Reply:
That should work, indeed. I would have to mostly rewrite `structured_to_unstructured` to account for it, and it seems like not enough of a good enough cost/benefit ratio for the effort, to me. I'd rather leave that to a separate PR if it's desired.

Reply:
As long as the contract of `structured_to_unstructured` allows us to make that change in a later release, I'm fine with not doing it now